ruby/examples/spec/register_user_spec.rb

```diff
@@ -0,0 +1,36 @@
+require_relative '../lib/register_user'
+
+describe RegisterUser do
+  subject do
+    register_user = described_class.new(gateway: FakeGateway.new)
+    register_user.register(RegisterUser::Request.new(email))
+  end
+
+  context 'when user created without errors' do
+    let(:email) { 'a@a.com' }
+    it { expect(subject.values).to include(email: email) }
+    it { expect(subject.errors).to be_empty }
+  end
+
+  context 'when user has errors' do
+    let(:email) { nil }
+    it { expect(subject.values).to include(email: email) }
+    it { expect(subject.errors).to_not be_empty }
+  end
+end
+
+describe RegisterUser::Gateway do
+  context 'when create user' do
+    subject { described_class.new.create_user(email: 'luke@cool.com') }
+
+    context 'and no errors' do
+      before { allow(Spree::User).to receive(:create) { double(errors: []) } }
```

Comment:
When I test Gateways I test the IO behaviour... So I would not Mock Spree...

Spree is a type we don't own, so we should not mock it.